README.md

```diff
@@ -198,8 +198,36 @@ A useful pattern here is to encapsulate the details of updating your elements in
 </script>
 ```
 
+_We could add a feature to make this even simpler, for custom-element cases like this. See [#101](https://github.com/valdrinkoshi/virtual-scroller/issues/101)._
+
 Note that in all these examples, the elements are recycled.
 
+#### Relation to template instantiation proposal
+
+The [template instantiation](https://github.com/w3c/webcomponents/blob/gh-pages/proposals/Template-Instantiation.md) proposal is a still-evolving idea for how to provide native updating of `<template>` elements to the web platform. This seems like exactly the kind of thing we'd want to use with `<virtual-scroller>`, once it becomes ready. For example, one could replace the above example with
+
+```html
+<virtual-scroller id="scroller">
+  <template>
+    <section>
+      <h1>{{name}}</h1>

```

Comment:
I'd update this example to use `{{contact.name}}` - which stresses another controversial point that the default processor needs to tackle, that is what context to pass into the template, e.g. 

```js
window.avatarBaseURL = 'http://my-cdn.com/contacts/id/';

scroller.updateElement = (section, contact, index) => {
  section.querySelector('h1').textContent = index + ') ' + contact.name;
  section.querySelector('img').src = avatarBaseURL + contact.id;
};
```
With template instantiation I'd expect to use:
```html
<virtual-scroller id="scroller">
  <template>
    <section>
      <h1>{{index}}) {{contact.name}}</h1>
      <img src="{{avatarBaseURL}}{{contact.id}}">
```
^ I'm not sure how the template instantiation would be able to get index and avatarBaseURL.
The virtual-scroller would have to pass the `contact` and `index` to the context of the template processor.

Reply:
After thinking about this, I think we should not change the example to show more controversial default template processor features, as it kind of distracts from the point. Instead, we can assume that the item source always contains all the necessary information. So for your example, I might do something like

```js
scroller.itemSource = new ItemSource({
  item(index) {
    return {
      index,
      avatarURL: avatarBaseURL + contacts[index].id,
      name: contacts[index].name
    };
  },
  getLength() {
    return contacts.length;
  }
});
```

I wonder if we should add some helper for this in the future... e.g. ItemSource.fromArrayWithMappingFunction() or something. Maybe I'll file a tracking issue?

Reply:
Sounds good